src/common/components/layouts/pages/inspect/tree/Tree.js

```diff
@@ -124,11 +134,13 @@ function Tree({ ruleSummaries, selectedCheck, setSelectedCheck, errorsMap, profi
     useEffect(() => {
         if (selectedCheck && selectedCheck !== prevSelectedCheck) {
             let ruleIndex = errorsMap[selectedCheck]?.ruleIndex;
-            if (ruleIndex !== expandedRule) {
-                setExpandedRule(ruleIndex);
+            const copyExpandedRule = JSON.parse(JSON.stringify(expandedRule));
```

Comment:
Reuse the method passed in props instead of code duplication (See my very first comment).
Note: `if (expandedRule.includes(ruleIndex)) { copyExpandedRule.splice(ruleIndex, 1, ruleIndex);` means that you set the rule open only if it's already opened. This will be fixed by reusing the method from Inspect component